internal/pkg/skuba/addons/cilium_test.go

```diff
@@ -0,0 +1,324 @@
+				config: tt.config,
+			}
+			got := renderContext.CiliumInitImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !matched {
+				t.Errorf("renderContext.CiliumInitImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderContext_CiliumOperatorImage(t *testing.T) {
+	type test struct {
+		name   string
+		addon  Addon
+		config AddonConfiguration
+		want   string
+	}
+	tests := []test{}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tests = append(
+			tests,
+			test{
+				name: "render Cilium Operator Image URL when cluster version is " + ver.String(),
+				addon: Addon{
+					addon:             kubernetes.Cilium,
+					templater:         renderCiliumTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetCiliumImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+				want: img.ImageRepository + "/cilium-operator:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
+			},
+		)
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			renderContext := renderContext{
+				addon:  tt.addon,
+				config: tt.config,
+			}
+			got := renderContext.CiliumOperatorImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !matched {
+				t.Errorf("renderContext.CiliumOperatorImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderContext_CiliumImage(t *testing.T) {
+	type test struct {
+		name   string
+		addon  Addon
+		config AddonConfiguration
+		want   string
+	}
+	tests := []test{}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tests = append(
+			tests,
+			test{
+				name: "render Cilium Image URL when cluster version is " + ver.String(),
+				addon: Addon{
+					addon:             kubernetes.Cilium,
+					templater:         renderCiliumTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetCiliumImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+				want: img.ImageRepository + "/cilium:[[:digit:]]{1,}.[[:digit:]]{1,}.[[:digit:]]{1,}[-rev[:digit:]{1,}]{0,1}",
+			},
+		)
+	}
+	for _, tt := range tests {
+		tt := tt // Parallel testing
+		t.Run(tt.name, func(t *testing.T) {
+			renderContext := renderContext{
+				addon:  tt.addon,
+				config: tt.config,
+			}
+			got := renderContext.CiliumImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !matched {
+				t.Errorf("renderContext.CiliumImage() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_ciliumCallbacks_beforeApply(t *testing.T) {
```

Comment:
If test cilium's `beforeApply`, is it possible to add more tests to increase the `beforeApply` test coverage?

Reply:
not able to pass in fake.client

Reply:
okay.